File: src/main/java/com/mycompany/basket_price/model/PriceBasket.java
/**
 * 
 */
package com.mycompany.basket_price.model;

import java.util.HashMap;
import java.util.Map;

/**
 * A basket of items 
 * 
 * @author colin
 *
 */
public class PriceBasket {

	// map of items in basket and how many of each item
	private Map<BasketItem, Integer> basketItems = null;
	
	/**
	 * Constructor
	 */
	public PriceBasket(){
		basketItems = new HashMap<>();
	}
	
	/**
	 * Puts an item into the basket
	 * 
	 * If there is already the same item in basket then increment the 
	 * quantity number
	 * 
	 * @param item to put in basket
	 */
	public void putItemInBasket(BasketItem item){
		basketItems.put(item, 1); //default 1 qty
	}
	
	/**
	 * Gets all the items in basket with the quantity
	 * 
	 * @return a map of items and their quantity in basket 
	 */
	public Map<BasketItem, Integer> getBasketItems(){
		return basketItems;
	}
}

File: src/main/kotlin/com/mycompany/basket_price/model/PriceBasket.kt
/*
 * |-------------------------------------------------
 * | Copyright © 2019 Colin But. All rights reserved.
 * |-------------------------------------------------
 */
package com.mycompany.basket_price.model

class PriceBasketK {

    private var basketItems : MutableMap<PriceBasketItemK, Int>? = null

    init {
        basketItems = HashMap()
    }

    fun putItemInBasket(newItemK: PriceBasketItemK) {
        for (item in basketItems!!.keys) {
            if (item.name == newItemK.name) {
                basketItems!![item] = basketItems!!.get(item)!!.toInt() + 1
                return
            }
        }
        basketItems!![newItemK] = 1
    }

    fun getBasketItems() : Map<PriceBasketItemK, Int> {
        return basketItems!!
    }

    fun basketContainItem(item: PriceBasketItemK) : Boolean {
        return basketItems!!.filterKeys { it.name == item.name }.isNotEmpty()
    }
}